### PR TITLE
refac: Ip refac v1

### PR DESCRIPTION
**Pull request by Macbeth98**
- Updated Readme by resolving etherscan spell mistakes in the test transactions links.
- Removed the usage of `Object.keys(scamSnifferDB.data.).filter(key => {})` by replacing it with `scamSnifferMap`. This makes the fetching of `scamDomains O(1) instead of O(n)` 
- Tested the Bot by running `npm test`. All tests are passing.
- Also manually tested the bot by running the given test transactions in the readme. All the transactions are returning findings except the last three of the Uniswap v3 attack. See the hashes below.
- **Note**: The below three txns are not returning any findings even with the freshly clone repo.

The Transactions are: 
1. [0x8f13bcbd56ef6c4ebdf1c18388ae4510be358b516aef4347b7d989b0340a1ae8](https://etherscan.io/tx/0x8f13bcbd56ef6c4ebdf1c18388ae4510be358b516aef4347b7d989b0340a1ae8)
2. [0x43337dadfd774ffdbb883f0935f1ba368d9fceb24a161e157cf4402e824dfbfd](https://etherscan.io/tx/0x43337dadfd774ffdbb883f0935f1ba368d9fceb24a161e157cf4402e824dfbfd)
3. [0x519802e340fe178bb573b6ad840a2eb56ba2638cffc5791860aa4af2fa05b398](https://etherscan.io/tx/0x519802e340fe178bb573b6ad840a2eb56ba2638cffc5791860aa4af2fa05b398)

### CODE REVIEW

File: ice-phishing/src/helper.js [hunk 1]
```diff
@@ -1617,6 +1617,35 @@ function checkObjectSizeAndCleanup(obj) {
   }
 }
 
+const populateScamSnifferMap = async (scamSnifferDB) => {
```
Comment (Vxatz) [inline ice-phishing/src/helper.js:1620]: @Macbeth98, is there a reason for using async/await on this?

Comment (Macbeth98) [reply ice-phishing/src/helper.js:1620]: I used async/await as I did `Promise.all()`. But now that I see again, this function and the other function also will not require async/await as there was no any asynchronous tasks/functions in those methods. 

I will update the code by removing the async/awaits

Comment (Macbeth98) [reply ice-phishing/src/helper.js]: This is done @Vxatz! The code is updated.

**Review by Vxatz — APPROVED** [review]
LGTM!

**Pull request merged**